DEVELOPMENT.md

```diff
@@ -1,40 +1,80 @@
 # Development
 
-## Testing via examples
-
-1. Add a new example project in `examples` folder. It is a regular NPM package with its own `package.json` and Cypress dev dependency.
-1. Add a corresponding `.github/workflows` YAML file that uses this action and runs inside `examples/X` working directory. The example should demonstrate the feature
-1. Add workflow badge to README if needed
-
-## Testing against another repo
-
-1. Create a new local branch for any development, for example `git checkout -b featureA`
-1. Update the source code in [index.js](index.js)
-1. Build `dist` file(s) using `npm run build`
-1. Commit any changed files, note the SHA of the commit
-1. Push the local branch to GitHub
-1. In a test repository, create a test branch, change the action to point at the new commit. Example
-
-```yml
-name: End-to-end tests
-on: push
-jobs:
-  cypress-run:
-    runs-on: 22.04
-    steps:
-      - uses: actions/checkout@v3
-      # use specific commit pushed to GitHub
-      - uses: cypress-io/github-action@<full commit SHA>
+This document describes topics useful to contributors to this repository. The repository's purpose is to provide a GitHub JavaScript action which is published to:
+
+1. npm's JavaScript Package Registry as [@cypress/github-action](https://www.npmjs.com/package/@cypress/github-action)
+2. GitHub's Marketplace as [cypress-io/github-action](https://github.com/marketplace/actions/cypress-io#cypress-iogithub-action--)
+
+You can read the GitHub document [Creating a JavaScript action](https://docs.github.com/en/actions/creating-actions/creating-a-javascript-action) for background information on how JavaScript actions for GitHub are created and how they work.
+
+## Providing fixes or features
+
+If you are submitting a Pull Request (PR) to provide a fix or feature for the action, the following is relevant to you:
+
+The main source code elements for the action are:
+- [action.yml](action.yml)
+- [index.js](index.js)
+- [src/ping.js](src/ping.js)
+
+The action runs from the [dist](dist) directory, which requires a `build` step to set up. This step uses [@vercel/ncc](https://github.com/vercel/ncc) to compile the code and modules into one file used for distribution. See below:
+
+1. If you are a new external contributor, then first fork the repository (see GitHub documentation [About forks](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/about-forks)).
+1. Create a new branch with a meaningful name e.g. `fix/my-bug` based on the current `master` branch.
+1. Make the necessary source code changes, including additions or changes to the [README.md](./README.md) documentation if parameters are added or affected.
+1. Ensure you are using [Node.js LTS](https://nodejs.org/en).
```

Comment:
Can we add a `.node-version` file specifying the development version so we are explicit when the develop version changes? typically `.nvmrc` files are used, but we have `.node-version` in several of our projects and this would align with that pattern.

Reply:
Sounds too to add in a follow-up. That makes sense if `n` uses `.node-version`. I always wondered where this came from but didn't look too deep into it. 

I personally use `nvm` and wrote a [custom zsh hook](https://github.com/emilyrohrbough/tips-and-tricks/blob/main/zsh-hook/load-node-version.sh) to handle changing and installing my node versions when I change directories on my machine.

I don't think having a standardized suggestion of a node version manager is needed for users to contribute. Knowing where to find the development version between projects is the main thing.

Reply:
@emilyrohrbough 
> Sounds too to add in a follow-up.

There is something missing in that sentence. I'm going to assume that a separate PR is OK though.

Would you accept using `.nvmrc` to hold the Node.js version in preference to `.node-version`? 
I believe that `nvm` is quite widely used and

```bash
nvm use
```

with no version specified will automatically pick up the version specified in `.nvmrc` and use it.

It's interesting to see your script for this purpose!

Reply:
Good with a followup, but for now lets go with `.node-version`. This is picked up by [`avn`](https://github.com/wbyoung/avn), which we suggest using out in the Cypress repo's Contribution doc: https://github.com/cypress-io/cypress/blob/develop/CONTRIBUTING.md#requirements.

Reply:
@emilyrohrbough 

- Unfortunately `avn` no longer works, so I submitted https://github.com/cypress-io/cypress/issues/26441 to get that clarified. I will hold back on submitting a `.node-version` here in the meantime. The instruction to use Node.js LTS is sufficient for the moment. Node.js 20 doesn't move to LTS until October 2023, so there is plenty of time to sort this issue out. Node.js 18 = current LTS is good until then.

Reply:
@emilyrohrbough 

- This thread is resolved in PR https://github.com/cypress-io/github-action/pull/901.